### PR TITLE
Add cashback app tutorial, Closes #236

### LINKED ISSUE #236: Example blockchain application Cashback
Parent: #203

Create section that explains the Hello World App in a detailed way.

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #236

### CODE REVIEW

File: examples.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 Welcome to the step-by-step guide of creating the Hello World application with Lisk Alpha SDK.
 A simple App, showcasing a minimal setup of a blockchain application with 1 [custom transaction](custom-transactions.md) type: the "Hello" transaction.
```
Comment (RachBLondon) [inline examples.md]:
```suggestion
A simple App, showcasing a minimal setup of a blockchain application with 1 [custom transaction](custom-transactions.md) type: the "HelloWorld" transaction.
```

File: examples.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
...
 
-The purpose of Hello World application is to explain how to use and how to implement custom transaction with the Lisk SDK. 
+The purpose of Hello World application is to explain how to use and how to implement custom transactions with the Lisk SDK.
```
Comment (RachBLondon) [inline examples.md]:
```suggestion
The purpose of Hello World application is to show how to implement custom transactions with the Lisk SDK. 
```

File: examples.md [hunk 3]
```diff
@@ -7,7 +7,7 @@
 Welcome to the step-by-step guide of creating the Hello World application with Lisk Alpha SDK.
 A simple App, showcasing a minimal setup of a blockchain application with 1 [custom transaction](custom-transactions.md) type: the "Hello" transaction.
 
-The purpose of Hello World application is to explain how to use and how to implement custom transaction with the Lisk SDK. 
+The purpose of Hello World application is to explain how to use and how to implement custom transactions with the Lisk SDK. 
 The implementation is saving the string value of the "hello" transaction's asset property to the asset property of the sender's account.
```
Comment (RachBLondon) [inline examples.md]:
```suggestion
The implementation is saving the string value of the "helloWorld" transaction's asset property to the asset property of the sender's account.
```

File: examples.md [hunk 4]
```diff
@@ -396,14 +397,238 @@ As next step, you can design a nice frontend application like [Lisk Explorer](ht
...
+}
+
+module.exports = CashbackTransaction;
+``` 
+
+> You find a version of [the complete `cashback_transaction.js` file](https://github.com/LiskHQ/lisk-sdk-examples/blob/development/cashback/transactions/cashback_transaction.js) on Github.
+
+#### TYPE
+
+Set the Cashback transaction TYPE to `11`.
```
Comment (RachBLondon) [inline examples.md:458]: Again should reserve more than 10 for Lisk?

File: examples.md [hunk 5]
```diff
@@ -396,14 +397,238 @@ As next step, you can design a nice frontend application like [Lisk Explorer](ht
...
 
+### 3. Create a new transaction type
+
+Now, we want to create a new [custom transaction type](custom-transactions.md) `CashbackTransaction`: 
 It extends the pre-existing transaction type `TransferTransaction`.
+In difference to the normal `TransferTransaction`, the Cashback transaction type pays out a 10% bonus reward to the sender of any Cashback transaction type.
+
+So e.g. if Alice sends 100 token to Bob as a Cashback transaction, Bob would receive the 100 token and Alice would receive additional 10 tokens as a cashback.
+
+> If you compare the methods below with the methods we implemented for the `HelloTransaction`, you will notice, that we implement less methods for the `CashbackTransaction`.
```
Comment (RachBLondon) [inline examples.md:435]:
```suggestion
> If you compare the methods below with the methods we implemented for the `HelloWorldTransaction`, you will notice, that we implement less methods for the `CashbackTransaction`.
```
If we want to move to the HelloWorld name

File: examples.md [hunk 6]
```diff
@@ -396,14 +397,238 @@ As next step, you can design a nice frontend application like [Lisk Explorer](ht
...
+Type `10` was used previously for the `HelloTransaction`, so we set it to `11`, but any other integer value (that is not already used by another transaction type) is a valid value.
+```js
+static get TYPE () {
+    return 11;
+}
+```
+
+#### applyAsset
+
+It sends the transaction amount to the recipient and substracts 1/10 of the transaction amount from the senders account balance
```
Comment (RachBLondon) [inline examples.md]:
```suggestion
The CashbackTransaction adds an inflationary 10% to senders account.
```

File: examples.md [hunk 7]
```diff
@@ -396,14 +397,238 @@ As next step, you can design a nice frontend application like [Lisk Explorer](ht
...
+Add the new transaction type to your application, by registering it to the application instance:
+
+```js
+//index.js
+const { Application, genesisBlockDevnet, configDevnet} = require('lisk-sdk'); // require application class, the default genesis block and the default config for the application
+const CashbackTransaction = require('./cashback_transaction'); // require the newly created transaction type 'CashbackTransaction'
+
+const app = new Application(genesisBlockDevnet, configDevnet); // create the application instance
+
+app.registerTransaction(11, CashbackTransaction); // register the 'CashbackTransaction'
```
Comment (RachBLondon) [inline examples.md:533]:
```suggestion
app.registerTransaction(CashbackTransaction); // register the 'CashbackTransaction' 
```
According @nazarhussain we can remove the number, is this correct @MaciejBaj ?

File: examples.md [hunk 8]
```diff
@@ -396,14 +397,238 @@ As next step, you can design a nice frontend application like [Lisk Explorer](ht
...
+Next, let's configure the application, to provide basic information about the app we are going to build:
+
+```js
+//index.js
+const { Application, genesisBlockDevnet, configDevnet } = require('lisk-sdk'); // require application class, the default genesis block and the default config for the application
+
+const app = new Application(genesisBlockDevnet, configDevnet); // create the application instance
+```
+
+In the `line 2`, we require the needed dependencies from the `lisk-sdk` package.
```
Comment (RachBLondon) [inline examples.md:420]:
```suggestion
On `line 2`, we require the needed dependencies from the `lisk-sdk` package.
```

File: examples.md [hunk 9]
```diff
@@ -396,14 +397,238 @@ As next step, you can design a nice frontend application like [Lisk Explorer](ht
...
+static get TYPE () {
+    return 11;
+}
+```
+
+#### applyAsset
+
+It sends the transaction amount to the recipient and substracts 1/10 of the transaction amount from the senders account balance
+
+Invoked as part of the `apply` step of the BaseTransaction and block processing.
```
Comment (RachBLondon) [inline examples.md:471]:
```suggestion
```
I think this sentence is a little confusing

File: examples.md (hunk 3 above)
Comment (MaciejBaj) [reply examples.md]: Let's keep "hello" now and wait for the possible feedback, if it's too complicated.

File: examples.md (hunk 9 above)
Comment (MaciejBaj) [reply examples.md:471]: There is an apply step on base transaction as well, applyAsset is being called as part of it.

**Review by RachBLondon — APPROVED** [review]

**Pull request merged**